central/scannerdefinitions/handler/handler_test.go

```diff
@@ -397,100 +511,103 @@ func (s *handlerTestSuite) TestServeHTTP_Offline_Get_V4() {
 	h.ServeHTTP(w, req)
 	s.Equal(http.StatusNotFound, w.Code)
 
-	url := "https://storage.googleapis.com/scanner-support-public/offline/v1/4.5/scanner-vulns-4.5.zip"
-	req, err := http.NewRequestWithContext(s.ctx, http.MethodGet, url, nil)
-	s.Require().NoError(err)
-	resp, err := http.DefaultClient.Do(req)
-	s.Require().NoError(err)
-	s.T().Cleanup(func() {
-		_ = resp.Body.Close()
-	})
-	s.Require().Equal(http.StatusOK, resp.StatusCode)
-
 	filePath := filepath.Join(s.T().TempDir(), "test.zip")
 	outFile, err := os.Create(filePath)
 	s.Require().NoError(err)
-	_, err = io.Copy(outFile, resp.Body)
+	_, err = io.Copy(outFile, newZipBuilder().
+		addFile(s, "scanner-defs.zip", "Scanner V2 content", []byte(content1)).
+		addFile(s, "scanner-v4-defs-4.5.zip", "Scanner V4 4.5", newZipBuilder().
+			addFile(s, "manifest.json", "Scanner V4 manifest", []byte(`{
+					  "version": "4.5",
+					  "release_versions": "some-release"
+					}`)).BuildBuffer(s).Bytes()).
+		addFile(s, "v4-definitions-v2.zip", "Scanner V4 v2", newZipBuilder().
+			addFile(s, "manifest.json", "Scanner V4 manifest", []byte(`{
+					  "version": "v2",
+					  "release_versions": "some-release"
+					}`)).
+			addFile(s, "vulnerabilities.zip", "Scanner V4 vulnerabilities", []byte(content2)).
+			addFile(s, "container-name-repos-map.json", "Scanner V4 repo-to-name map", []byte(`{}`)).
+			addFile(s, "repository-to-cpe.json", "Scanner V4 repo-to-cpe map", []byte(`{}`)).
+			BuildBuffer(s).Bytes()).
+		BuildBuffer(s))
 	s.Require().NoError(err)
 	utils.IgnoreError(outFile.Close)
 
 	// Upload offline vulns, directly.
 	err = s.mockHandleZipContents(filePath)
 	s.Require().NoError(err)
 
-	// This will fail because 4.5.0 uses the multi-bundle ZIP format.
-	req = s.getRequestVersion("4.5.0")
-	w = httptest.NewRecorder()
-	h.ServeHTTP(w, req)
-	s.Equal(http.StatusNotFound, w.Code)
+	s.T().Run("get 4.5", func(t *testing.T) {
+		req = s.getRequestVersion("4.5.0")
+		w = httptest.NewRecorder()
+		h.ServeHTTP(w, req)
+		// This fails on release builds because checks don't happen on dev builds.
+		if buildinfo.ReleaseBuild {
+			s.Equalf(http.StatusNotFound, w.Code, "body: %s", w.Body.String())
+		} else {
+			s.Equal(http.StatusOK, w.Code, "body: %s", w.Body.String())
```

Comment:
we should be sure to check the contents match, too (or at least do some kind of contents sanity check)

Reply:
7c786ab1ec